api/src/exam-environment/routes/exam-environment.ts

```diff
@@ -0,0 +1,547 @@
+/* eslint-disable jsdoc/require-returns, jsdoc/require-param */
+import { type FastifyPluginCallbackTypebox } from '@fastify/type-provider-typebox';
+import { type FastifyInstance, type FastifyReply } from 'fastify';
+import jwt from 'jsonwebtoken';
+
+import * as schemas from '../schemas';
+import { mapErr, syncMapErr, UpdateReqType } from '../../utils';
+import { JWT_SECRET } from '../../utils/env';
+import {
+  checkAttemptAgainstGeneratedExam,
+  checkPrerequisites,
+  constructUserExam,
+  generateExam,
+  userAttemptToDatabaseAttemptQuestionSets,
+  validateAttempt
+} from '../utils/exam';
+import { ERRORS } from '../utils/errors';
+
+/**
+ * Wrapper for endpoints related to the exam environment desktop app.
+ *
+ * Requires exam environment authorization token to be validated.
+ */
+export const examEnvironmentValidatedTokenRoutes: FastifyPluginCallbackTypebox =
+  (fastify, _options, done) => {
+    // TODO: Is there any reason to delete the token without generating a new one?
+    fastify.post(
+      '/exam-environment/exam/generate',
+      {
+        schema: schemas.examEnvironmentPostExamGenerate
+      },
+      postExamGenerateHandler
+    );
+    fastify.post(
+      '/exam-environment/exam/attempt',
+      {
+        schema: schemas.examEnvironmentPostExamAttempt
+      },
+      postExamAttemptHandler
+    );
+    fastify.post(
+      '/exam-environment/screenshot',
+      {
+        schema: schemas.examEnvironmentPostScreenshot
+      },
+      postScreenshotHandler
+    );
+    done();
+  };
+
+/**
+ * Wrapper for endpoints related to the exam environment desktop app.
+ *
+ * Does not require exam environment authorization token to be validated.
+ */
+export const examEnvironmentOpenRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.post(
+    '/exam-environment/token/verify',
```

Comment:
The point is, when something is pasted in the app, it is verified with this endpoint that it exists. That way, the Camper does not find out as they are about to take the exam something is wrong.

Reply:
Just to confirm: the idea is that may not be signed into their FCC account (i.e. via `jwt_access_token`), so they can't simply re-create the token before starting the exam?

Reply:
Yes, that is one of the reasons to have this.

The other is anytime anything _protected_ happens on the app, the token needs to be checked, because of the way we are storing the keys.